### PR TITLE
C#: update MaD sink kinds

### PR DESCRIPTION
**Pull request by jcogs33**
### Description
This PR updates some C# MaD sink kind names in order to align across languages. 
The following sink kind names are updated:
| Old kind name      | New kind name |
| ----------- | ----------- |
| `code`   | `code-injection`        |
| `sql`      | `sql-injection`       |
| `html`   | `html-injection`        |
| `xss`   | `js-injection`        |
| `remote`   | `file-content-store`        |

Commit-by-commit review might be best.

(related Java PR: https://github.com/github/codeql/pull/12916)

### Consideration
- Let me know if any of the updates will be problematic for C# (especially the `xss` to `js-injection` and `remote` to `file-content-store` updates).
- Let me know if I should add a change note. (I'm currently assuming it is not needed since C# custom data extensions are not in beta, but let me know if I should add a change note anyways.)

### CODE REVIEW

File: csharp/documentation/library-coverage/cwe-sink.csv [hunk 1]
```diff
@@ -1,2 +1,2 @@
 CWE,Sink identifier,Label
-CWE-079,html xss,Cross-site scripting
\ No newline at end of file
+CWE-079,html-injection js-injection,Cross-site scripting
```
Comment (michaelnebel) [inline csharp/documentation/library-coverage/cwe-sink.csv:2]: A couple of clarifying questions
(1) Do you know what this file is being used for (where is it consumed)? :-)
(2) Could you remind me what "js" is an abbreviation for in "js-injection"?

**Review by michaelnebel — COMMENTED** [review]
Thank you for doing this!

I think it makes sense to add a change note in case this is used by the field team.

File: csharp/documentation/library-coverage/cwe-sink.csv (hunk 1 above)
Comment (jcogs33) [reply csharp/documentation/library-coverage/cwe-sink.csv:2]: (1) It is used to create the `CWE-079 Cross-site scripting` column in [coverage.rst](https://github.com/github/codeql/blob/5a1c001f07e55450e458959d960ce196ef5af10f/csharp/documentation/library-coverage/coverage.rst).

(2) "js" is an abbreviation for JavaScript. I'll send you a link in Slack to the conversation that discussed the `xss` to `js-injection` change. Let me know if you think this change doesn't make sense for C# after all.

**Review by michaelnebel — APPROVED** [review]
Thank you!

**Comment by michaelnebel** [comment]
@jcogs33 : It is probably a good idea to run DCA before merging.

**Comment by jcogs33** [comment]
@michaelnebel There were a few failures in the DCA run. Do those look like a cause for concern?

**Comment by michaelnebel** [comment]
> @michaelnebel There were a few failures in the DCA run. Do those look like a cause for concern?

That is not related to your changes (looks like transient failures). The analyzed projects don't show any alerts diff - so I think you are changes are good to go :-)
Thank you for doing this for C#!

**Pull request merged**